CMakeLists.txt

```diff
@@ -109,42 +103,8 @@ if(BUILD_BENCHMARKS)
   add_subdirectory(benchmarks)
 endif(BUILD_BENCHMARKS)
 
-###################################################################################################
-# - library targets -------------------------------------------------------------------------------
-
-add_library(rmm SHARED "${CNMEM_SOURCE_DIR}/cnmem.cpp")
-
-# override rpath for rmm
-set_target_properties(rmm PROPERTIES BUILD_RPATH "\$ORIGIN")
-
-target_include_directories(rmm PUBLIC
-  "${THRUST_INCLUDE_DIR}"
-  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-  "$<INSTALL_INTERFACE:include>"
-  )
-
-# In order to find cnmem.h while building the library, we need this.
-# For library users it is replaced by a copy of the same cnmem.h under rmm/detail
-target_include_directories(rmm PRIVATE
-  "${CNMEM_INCLUDE_DIR}"
-  )
 
 ###################################################################################################
-# - link libraries --------------------------------------------------------------------------------
-
-if(CUDA_STATIC_RUNTIME)
-  message(STATUS "Enabling static linking of cudart")
-  target_link_libraries(rmm PUBLIC CUDA::cudart_static)
-else()
-  target_link_libraries(rmm PUBLIC CUDA::cudart)
-endif(CUDA_STATIC_RUNTIME)
```

Comment:
I'm not sure that the static runtime was used even before, so that can probably go away. But the `target_link_libraries` should also remain. If an executable "links" rmm (nothing to link, so that just means "uses"), it'll have to transitively pull in cudart, both the actual library, as well as the include paths that go with it. That's what this `target_link_libraries` will do.

Reply:
There's nothing wrong fundamentally with a header-only library linking (using) a static library in general, btw. So whether `CUDA_STATIC_RUNTIME` should be continued to be supported depends on whether anyone is using it, which I don't know.

Reply:
Same goes for this. I added this back, but I seem to still need it inside the tests and benchmarks cmakelists, which we didn't before. I was hoping to reduce duplication, not create duplication.

Reply:
Figured it out. Have to "link" rmm even though it's not a binary library.  Weird.

Reply:
They really should call it a module dependency or something. Using the language of the linker when you aren't linking is just confusing.

Reply:
Done.

Reply:
> They really should call it a module dependency or something. Using the language of the linker when you aren't linking is just confusing.

I agree, in principle. It's basically a historic artifact that it says `target_link_libraries`, where "link" now just means "use", but the old language is still used.